### PR TITLE
Added severity score vector

### PR DESCRIPTION
**Pull request by Cheshiriks**
Severity score vector is a group of indicators that show the importance of vulnerability. They are necessary to calculate the criticality rating.

Снимок

### CODE REVIEW

**Review by sanyavertolet — COMMENTED** [review]
I do understand nothing about what you've done. Maybe add some tips on what the vector is? Why do you need to provide it and so on. 

I mean it might be something important, but I don't know what's that

**Comment by akuleshov7** [comment]
> I do understand nothing about what you've done. Maybe add some tips on what the vector is? Why do you need to provide it and so on.
> 
> I mean it might be something important, but I don't know what's that

Ор

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt [hunk 1]
```diff
@@ -41,6 +42,7 @@ data class VulnerabilityDto(
     val creationDateTime: LocalDateTime? = null,
     val lastUpdatedDateTime: LocalDateTime? = null,
     val tags: Set<String> = emptySet(),
+    val critical: String = "",
```
Comment (akuleshov7) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt:45]: why critical? We discussed it, it should be the same to COSV

Comment (akuleshov7) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt:45]: severity

**Comment by akuleshov7** [comment]
And also delete the field from Metadata

**Comment by sanyavertolet** [comment]
> > I do understand nothing about what you've done. Maybe add some tips on what the vector is? Why do you need to provide it and so on.
> > I mean it might be something important, but I don't know what's that
> 
> Ор

Yep, but to my mind frontend should be more clear - some tooltips or a guide page to vuln creation

**Review by akuleshov7 — APPROVED** [review]
Please fix build and lgtm

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/cosv/CosvMetadata.kt [hunk 2]
```diff
@@ -31,7 +30,6 @@ class CosvMetadata(
     var cosvId: String,
     var summary: String,
     var details: String,
-    var severity: String?,
```
Comment (akuleshov7) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/cosv/CosvMetadata.kt:34]: I think this change should affect database

**Review by nulls — APPROVED** [review]

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/cosv/CosvMetadata.kt (hunk 2 above)
Comment (Cheshiriks) [reply save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/cosv/CosvMetadata.kt:34]: Fixed

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputForms.kt [hunk 3]
```diff
@@ -10,6 +10,7 @@ import com.saveourtool.save.validation.*
 
 private const val URL_PLACEHOLDER = "https://example.com"
 private const val EMAIL_PLACEHOLDER = "test@example.com"
+private const val SEVERITY_VECTOR_PLACEHOLDER = "CVSS:3.1/AV:_/AC:_/PR:_/UI:_/S:_/C:_/I:_/A:_"
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputForms.kt:13]: So no tooltip/popover/other explanation is not present, is it?

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt [hunk 4]
```diff
@@ -23,6 +23,8 @@ import kotlinx.serialization.Serializable
  * @property creationDateTime [LocalDateTime] of creation
  * @property lastUpdatedDateTime [LocalDateTime] of last updating
  * @property tags
+ * @property critical
```
Comment (nulls) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt:26]:
```suggestion
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputForms.kt (hunk 3 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/inputform/InputForms.kt:13]: Added

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt (hunk 4 above)
Comment (Cheshiriks) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/entities/vulnerability/VulnerabilityDto.kt]: Fixed

**Review by sanyavertolet — APPROVED** [review]

**Pull request merged**